codalab/rest/bundle.py

```diff
@@ -61,6 +65,40 @@ def get_blob(uuid, path=''):
     return fileobj
 
 
+@put('/bundles/<uuid:re:%s>/contents/blob/' % spec_util.UUID_STR,
+     apply=AuthenticatedPlugin())
+def update_blob(uuid):
+    """
+    Update the contents of the given running or uploading bundle.
+
+    Accepts the filename as a query parameter, used to determine whether the
+    upload contains an archive.
+
+    Bundle state is updated to READY after upload is complete if the |finalize|
+    query parmater is nonzero.
+    """
+    bundle = local.model.get_bundle(uuid)
+    if not check_bundle_have_run_permission(local.model, request.user.user_id, bundle):
```

Comment:
Change to 

    check_bundles_have_all_permission(local.model, request.user, [uuid])
    bundle = local.model.get_bundle(uuid)

run permission is a bit simpler and really reserved for the worker.